applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_bfecc.h

```diff
@@ -0,0 +1,385 @@
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief Initialize the BFECC forward substep
+     * This method is intended to implement all the operations required before each BFECC initial forward substep
+     */
+    void InitializeBFECCForwardSubstep() override
+    {
+        BaseType::InitializeBFECCForwardSubstep();
+        StashDiffusiveConstants();
+        this->InitializeEverySubstep();
+    };
+
+    /**
+     * @brief Initialize the BFECC backward substep
+     * This method is intended to implement all the operations required before each BFECC backward substep
+     */
+    void InitializeBFECCBackwardSubstep() override
+    {
+        BaseType::InitializeBFECCBackwardSubstep();
+        this->InitializeEverySubstep();
+    };
+
+    /**
+     * @brief Initialize the BFECC final substep
+     * This method is intended to implement all the operations required before each BFECC final substep
+     */
+    void InitializeBFECCFinalSubstep() override
+    {
+        BaseType::InitializeBFECCFinalSubstep();
+        PopDiffusiveConstants();
+        this->InitializeEverySubstep();
+    };
+
+    /**
+     * @brief Initialize the BFECC forward substep
+     * This method is intended to implement all the operations required before each BFECC initial forward substep
+     */
+    void FinalizeBFECCForwardSubstep() override
+    {
+        BaseType::FinalizeBFECCForwardSubstep();
+        this->FinalizeEverySubstep();
+    };
+
+    /**
+     * @brief Finalize the BFECC backward substep
+     * This method is intended to implement all the operations required before each BFECC backward substep
+     */
+    void FinalizeBFECCBackwardSubstep() override
+    {
+        BaseType::FinalizeBFECCBackwardSubstep();
+        this->FinalizeEverySubstep();
+    };
+
+    /**
+     * @brief Finalize the BFECC final substep
+     * This method is intended to implement all the operations required before each BFECC final substep
+     */
+    void FinalizeBFECCFinalSubstep() override
+    {
+        BaseType::FinalizeBFECCFinalSubstep();
+        PopDiffusiveConstants();
+        this->FinalizeEverySubstep();
+    };
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    struct Stash {
+        double conductivity = 0.0;
+        double dynamic_viscosity = 0.0;
+    } mDiffusionStash;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    void StashDiffusiveConstants()
+    {
+        auto& model_part = BaseType::GetModelPart();
+        if(model_part.NumberOfElements() == 0) return;
```

Comment:
I think that an error should be thrown in this case.

Reply:
This may happen in MPI, for instance, without being erroneous.

Reply:
But there is no MPI support yet. I think it's better to throw an error for the moment. In any case, up to you.

Reply:
Ok, I'll put an error message with a coment.